src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -255,9 +273,109 @@ public virtual async Task<Overture.ServiceModel.Orders.Order> UpdateOrderCustome
                 ShipmentsNotes = shipmentsNotes
             });
 
+            if (order.Cart.PropertyBag.TryGetValue("PickedItems", out var pickedItemsObject))
+            {
+                var pickedItemsList = ComposerJsonSerializer.Deserialize<List<PickedItemViewModel>>(pickedItemsObject.ToString());
+                var shipment = viewModel.Shipments.First();
+                shipment.LineItems = await ProcessPickedLineItems(pickedItemsList, shipment.LineItems, getOrderParam.CultureInfo).ConfigureAwait(false);
+            };
+
             return viewModel;
         }
 
+        protected virtual async Task<List<LineItemDetailViewModel>> ProcessPickedLineItems(List<PickedItemViewModel> pickedItemsList,
```

Comment:
I think it could be better to rename this into ProcessPickedLineItems**Async** since it will be available not only in this class

Reply:
ok